src/modules/azure.rs

```diff
@@ -0,0 +1,89 @@
+use dirs::home_dir;
+use encoding_rs::*;
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read};
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use super::{Context, Module, RootModuleConfig};
+
+type JValue = serde_json::Value;
+
+use crate::configs::azure::AzureConfig;
+use crate::formatter::StringFormatter;
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("azure");
+    let config = AzureConfig::try_load(module.config);
+
+    let subscription_name = get_azure_subscription_name()?;
+
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter.map(|variable| match variable {
+            "subscription" => Some(subscription_name.to_string()),
+            _ => None,
+        })
+    } else {
+        log::warn!("Error parsing format string in `azure.format`");
+        return None;
+    };
+
+    module.set_segments(formatter.parse(None));
+
+    module.get_prefix().set_value("");
+    module.get_suffix().set_value("");
+
+    Some(module)
+}
+
+fn get_azure_config_file_location() -> Option<PathBuf> {
+    env::var("AZURE_CONFIG_DIR")
+        .ok()
+        .and_then(|path| PathBuf::from_str(&path).ok())
```

Comment:
I think there is an implicit allocate + copy here which you an avoid by using `PathBuf::from(path)` instead.